virtualization.go

```diff
@@ -198,6 +211,9 @@ func (v *VirtualMachine) CanRequestStop() bool {
 }
 
 // CanStop returns whether the machine is in a state that can be stopped.
+//
+// This is only supported on macOS 12 and newer, false will always be returned
+// on older versions.
 func (v *VirtualMachine) CanStop() bool {
```

Comment:
I will add some changes to `CanStop`/... to handle older macOS from the go code.

Regarding your exception suggestion, the `@available` check has to be `if (@available(...)`, it cannot be negated, see https://stackoverflow.com/questions/46388490/negating-objective-cs-available-keyword , so this would become 
```
if (@available(macOS 12, *)) {
} else {
  [[NSException exceptionWithName:@"UnhandledException" reason:@"bug" userInfo:nil] raise];
}
// write original code here.
```

I think I'll wrap the code above in a macro. It does not look very nice ^^

```
#define THROW_IF_MACOS_OLDER_THAN(version) \
    if (@available(macOS version, *)) { \
    } else { \
        [[NSException exceptionWithName:@"UnhandledException" reason:@"bug" userInfo:nil] raise]; \
    }
```

and then 
```
THROW_IF_MACOS_OLDER_THAN(12)
// write original code here.
```

Reply:
LGTM :+1:

Reply:
There is one big limitation with this. If I use this code:
```
void setDirectorySharingDevicesVZVirtualMachineConfiguration(void *config, void *directorySharingDevices)
{
    THROW_IF_MACOS_OLDER_THAN(12)

    [(VZVirtualMachineConfiguration *)config setDirectorySharingDevices:[(NSMutableArray *)directorySharingDevices copy]];
}
```

Then when building for macOS 11 (`CGO_CFLAGS=-mmacosx-version-min=11.0`) I get this warning:
```
virtualization.m:277:46: warning: 'setDirectorySharingDevices:' is only available on macOS 12.0 or newer [-Wunguarded-availability-new]
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/Virtualization.framework/Headers/VZVirtualMachineConfiguration.h:116:79: note: 'setDirectorySharingDevices:' has been marked as being introduced in macOS 12.0 here, but the deployment target is macOS 11.0.0
virtualization.m:277:46: note: enclose 'setDirectorySharingDevices:' in an @available check to silence this warning
```


The code can be written this way to use an exception and to remove the warning:
```
void setDirectorySharingDevicesVZVirtualMachineConfiguration(void *config, void *directorySharingDevices)
{
    if (@available(macOS 12, *)) {
        [(VZVirtualMachineConfiguration *)config setDirectorySharingDevices:[(NSMutableArray *)directorySharingDevices copy]];
    }  else {
        [[NSException exceptionWithName:@"UnhandledException" reason:@"bug" userInfo:nil] raise]; \
    }
}
```
I'd prefer to go with this code. It is very convenient and useful when the compiler can detect incompatibilities for you. It warns you when you forgot @available checks in new code.